app/components/embed/legacy_media_component.rb

```diff
@@ -16,8 +16,12 @@ def include_transcripts
     end
 
     def primary_files_count
-      purl_object.contents.count do |resource|
-        resource.primary_file.present?
+      resources_with_primary_file.count
+    end
+
+    def resources_with_primary_file
+      @resources_with_primary_file ||= purl_object.contents.select do |purl_resource|
+        purl_resource.primary_file.present?
       end
     end
```

Comment:
Naïvely, this code looks similar (identical?) to lines 22-26 in `app/components/embed/media_with_companion_windows_component.rb`. Do you think it's worth considering extracting them into a model or a service class, or do you expect these methods to diverge? (Alternatively, maybe there's nothing to do because the legacy component is not long for this world?)

Reply:
Yeah, the legacy component is supposed to go away, so I don't want to build connections between the two.